jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java

```diff
@@ -0,0 +1,161 @@
+
+    private void appendRow(final StringBuilder builder, final ResultSet resultSet,
+            final ResultSetMetaData resultSetMetadata) throws SQLException {
+        final int columnCount = resultSetMetadata.getColumnCount();
+        builder.append(" (");
+        for (int i = 1; i <= columnCount; ++i) {
+            final String columnName = resultSetMetadata.getColumnName(i);
+            if (i > 1) {
+                builder.append(", ");
+            }
+            appendColumnValue(builder, resultSet, columnName, resultSetMetadata.getColumnType(i));
+        }
+        builder.append(")");
+    }
+
+    private void appendColumnValue(final StringBuilder builder, final ResultSet resultSet, final String columnName,
+            final int type) throws SQLException {
+        switch (type) {
+        case Types.BIGINT:
+            builder.append(resultSet.getInt(columnName));
+            break;
+        case Types.NUMERIC:
+        case Types.DOUBLE:
+            builder.append(resultSet.getDouble(columnName));
+            break;
+        case Types.BOOLEAN:
+            builder.append(resultSet.getBoolean(columnName));
+            break;
+        case Types.DATE:
+            builder.append("'");
+            builder.append(castDate(resultSet.getString(columnName)));
+            builder.append("'");
+            break;
+        case Types.TIMESTAMP:
+            builder.append("'");
+            builder.append(castTimestamp(resultSet.getString(columnName)));
+            builder.append("'");
+            break;
+        case Types.VARBINARY:
+        case Types.TIME:
+        case Types.VARCHAR:
+        default:
+            builder.append("'");
+            builder.append(resultSet.getString(columnName));
+            builder.append("'");
+            break;
+        }
+    }
+
+    private String castTimestamp(final String timestampToCast) {
+        final String[] splitTimestamp = getSplitTimestamp(timestampToCast);
+        final StringBuilder builder = new StringBuilder();
+        builder.append(castDate(splitTimestamp[0]));
+        builder.append(" ");
+        builder.append(castTime(splitTimestamp[1]));
+        return builder.toString();
+    }
+
+    private String[] getSplitTimestamp(final String timestampToCast) {
+        if (timestampToCast.contains("T")) {
+            return timestampToCast.split("T");
+        } else {
+            return timestampToCast.split(" ");
+        }
+    }
+
+    private String castTime(final String timeToCast) {
+        final Matcher matcher = TIME_PATTERN.matcher(timeToCast);
+        if (matcher.matches()) {
+            final int hour = Integer.parseInt(matcher.group(1));
+            final int minute = Integer.parseInt(matcher.group(2));
+            final int second = Integer.parseInt(matcher.group(3));
+            final String milliseconds = matcher.group(4);
+            if (milliseconds != null) {
+                final int millisecondsInt = Integer.parseInt(milliseconds);
```

Comment:
What you get from BigQuery are at the finest resolution are nano seconds.

Reply:
fixed